rm-datasource/src/main/java/org/apache/seata/rm/datasource/xa/ConnectionProxyXA.java

```diff
@@ -93,6 +94,15 @@ public void init() {
             if (!currentAutoCommitStatus) {
                 throw new IllegalStateException("Connection[autocommit=false] as default is NOT supported");
             }
+            Integer transactionTimeout = RootContext.getTimeout();
+            if (transactionTimeout == null) {
+                transactionTimeout = DefaultValues.DEFAULT_GLOBAL_TRANSACTION_TIMEOUT;
+            }
+            timeout = Math.max(BRANCH_EXECUTION_TIMEOUT, transactionTimeout);
+            branchBeginTime = RootContext.getBranchBeignTime();
```

Comment:
I don't see this branchBeginTime value being passed from the downstream service, and if it's not passed, it's no different from the previous function. Or what you're trying to say is timeout is the timeout of a branch transaction not the timeout of a global transaction?